main.py

```diff
@@ -448,6 +470,330 @@ def move_methods_workaround(dex_filename, dex_filename_last, in_dir, out_dir, de
+    #
+    # Copy the patched file to the output folder
+    print_(" *** Copying the patched file to the output folder...")
+    safe_copy(os.path.join(TMP_DIR, "framework.jar"), os.path.join(OUTPUT_PATH, "framework.jar"))
+    #
+    # Clean up "framework.jar" disassembly folder.
+    print_(" *** Deleting temporary directory:", TMP_DIR+"/smali-classes")
+    deltree(TMP_DIR+"/smali-classes")
+    #
+    # Push patched file to device if selected.
+    if mode == 1:
+        enable_device_writing(SELECTED_DEVICE)
+        # Push to device
+        print_(" *** Pushing changes to the device...")
+        try:
+            if not DEBUG_PROCESS:
+                output = subprocess.check_output([DEPS_PATH["adb"], "-s", SELECTED_DEVICE, "push", "framework.jar", "/system/framework/framework.jar"], stderr=subprocess.STDOUT)
+                debug(output.decode("utf-8").rstrip())
+        except subprocess.CalledProcessError as e:
+            output = e.output.decode("utf-8")
+            debug(output.strip())
+            if e.returncode == 1 and "No space left on device" in output:
+                warning("Pushing has failed, we will retry from the recovery.")
+                subprocess.check_call([DEPS_PATH["adb"], "-s", SELECTED_DEVICE, "reboot", "recovery"])
+                subprocess.check_call([DEPS_PATH["adb"], "-s", SELECTED_DEVICE, "wait-for-device"])
+                enable_device_writing(SELECTED_DEVICE)
+                subprocess.check_output([DEPS_PATH["adb"], "-s", SELECTED_DEVICE, "push", "framework.jar", "/system/framework/framework.jar"])
+            else:
+                raise
+        # Kill ADB server
+        subprocess.check_call([DEPS_PATH["adb"], "kill-server"])
+
+
+
+#
+# FUNCTION - main_patch_services_jar
+#
+def main_patch_services_jar():
+    print_(" *** Decompressing services.jar ...")
+    decompress("services.jar", "services/")
+    #
+    # Disassemble it
+    print_(" *** Disassembling services classes...")
+    smali_to_search = "com/android/server/pm/PackageManagerService.smali"
+    smali_folder, dex_filename, dex_filename_last = find_smali(smali_to_search, "services/", DEVICE_SDK)
+    #
+    # Check the existence of the file to patch
+    if smali_folder is None:
+        print_(os.linesep+"ERROR: The services.jar classes file to patch cannot be found, please report the problem.")
+        exit_now(82)
+    to_patch = smali_folder+smali_to_search
+    #
+    # Do the injection
+    print_(" *** Patching services...")
+    f = open(to_patch, "r")
+    old_contents = f.readlines()
+    f.close()
+    #
+    f = open(SCRIPT_DIR+"/patches/allowsdwrite.smali", "r")
+    allowsdwrite = f.readlines()
+    f.close()
+    #
+    # services_classes - Add allowsdwrite method
+    #       for testing - \smali-classes\com\android\server\pm
+    i = 0
+    contents = []
+    already_patched = False
+    in_function = False
+    required_line_before = False
+    right_line = False
+    start_of_line = None
```

Comment:
__[flake8]__
* [F841] local variable 'start_of_line' is assigned to but never used

<!-- Comment by [SideCI](https://sideci.com) -->